commit 6f851174fe92c077e316005612e7246819434e0e
Author: Max Horn <max@quendi.de>
Date:   2022-09-14 00:12:02 +0200

    kernel: workaround GCC 12 warning
    
    With GCC 12, we get a spurious warning indicating that `buf` in `GetStr`
    may be used uninitialized. Weirdly, it does not complain about the almost
    identical code in e.g. `GetPragma`.

diff --git a/src/scanner.c b/src/scanner.c
index f5dbbcc04..23adc944e 100644
--- a/src/scanner.c
+++ b/src/scanner.c
@@ -653,7 +653,7 @@ static Char GetEscapedChar(ScannerState * s)
 static Char GetStr(ScannerState * s, Char c)
 {
     Obj  string = 0;
-    Char buf[1024];
+    Char buf[1024] = "";
     UInt i = 0;
 
     while (c != '"' && c != '\n' && c != '\377') {
